api/envoy/extensions/load_balancing_policies/override_host/v3/override_host.proto

```diff
@@ -0,0 +1,77 @@
+// [#extension: envoy.load_balancing_policies.override_host]
+
+// Configuration for the Override Host Load Balancing policy.
+//
+// This policy allows endpoint picking to be implemented in an external to Envoy
+// component. For example an ext_proc RPC to a service that implements k8s proposal for AI gateway inferences extensions
+// https://github.com/kubernetes-sigs/gateway-api-inference-extension/tree/main/docs/proposals/004-endpoint-picker-protocol
+// can provide hosts for serving a request using Override Host load balancing policy.
+//
+// This policy uses selected endpoints from either request headers or request metadata.
+//
+// The primary host override source must specify a single ``IP:port`` value in either the header or the
+// metadata. For example ``10.0.0.5:8080`` or ``[2600:4040:5204::1574:24ae]:80``. The IPv6 address is enclosed in square brackets.
+//
+// For example, to support k8s gateway inference extensions, which uses the ``x-gateway-destination-endpoint`` header or metadata value value under the "envoy.lb" key for the primary endpoint,
+// the Override Host load balancing policy should be configured in the following way:
+//
+// .. code-block:: yaml
+//
+//    primary_host_sources:
+//      - header: "x-gateway-destination-endpoint"
+//        metadata:
+//          key: "envoy.lb"
+//          path:
+//          - key: "x-gateway-destination-endpoint"
+//
+// If neither header nor metadata is present or there were errors parsing header or metadata values the
+// specified fallback load balancing policy is used. This allows load balancing to degrade to a
+// a built in policy (i.e. Round Robin) in case external endpoint picker fails.
+//
+// See the :ref:`load balancing architecture
+// overview<arch_overview_load_balancing_types>` for more information.
+//
+message OverrideHost {
+  message OverrideHostSource {
+    // The header to get the override host addresses. One of the header or metadata field must be set.
+    string header = 1
+        [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
+
+    // The metadata key to get the override host addresses from the request dynamic metadata. If
+    // set this field then it will take precedence over the header field.
+    // One of the header or metadata field must be set.
+    type.metadata.v3.MetadataKey metadata = 2;
+  }
+
+  // A list of host sources to get the override host addresses from. The first one who provides
+  // the addresses will be used, no matter if the addresses are valid or not. This ensure the first exist
+  // addresses source will be used.
+  repeated OverrideHostSource primary_host_sources = 1 [(validate.rules).repeated = {min_items: 1}];
+
+  // A list of host sources to get the fallback host addresses from. The first one who provides
+  // the addresses will be used, no matter if the addresses are valid or not.
+  // [#not-implemented-hide:]
+  repeated OverrideHostSource fallback_host_sources = 2;
```

Comment:
I didn't see the implementation, but I guess the fallback host source will contains a list of addresses, like `1.2.3.4:90,1.2.3.5:90`, then I think we can use single host source for primary and fallback? The first address will be used at first try and second address will be used at second try.

Reply:
See https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/761

Reply:
You still need two configurations for cases where an implementation that uses original_dst is transitioning to the override_host policy.

Reply:
> You still need two configurations for cases where an implementation that uses original_dst is transitioning to the override_host policy.

Yeah, the migration is another problem. The EPP may need to aware whether the proxy provide the fallback support and use different format.

But anyway, although the EPP is one of our design target, I will still hope we can treat this new extension as common feature and the EPP is only one usage of the new extension. That's say, we may needn't to consider the backward compatiblity of EPP self here.

Reply:
And no matter how the EPP implement the fallback, we can always treat the header value or metadata value (from the specified `host_sources`) as list of addresses and itrate them when finding a valid host. (The single address case could be treat as list which contains only one element.)

Reply:
Ok, sounds good